R/sem.R

```diff
@@ -239,17 +242,39 @@ checkLavaanModel <- function(model, availableVars) {
 
     # create options
     lav_args <- lavopts
-    syntax   <- .semTranslateModel(options[["models"]][[i]][["syntax"]], dataset)
-    lav_args[["model"]] <- syntax
+    originalSyntax   <- .semTranslateModel(options[["models"]][[i]][["syntax"]], dataset)
+    if(options[["group"]] == "")
+      syntaxTable <- lavaan::lavaanify(originalSyntax)
+    if (options[["group"]] != "") {
+      fit <- lavaan::sem(model = originalSyntax, data = dataset, group = options[["group"]])
+      syntaxTable <- lavaan::parTable(fit)
+    }
+
+    if(nrow(syntaxTable[syntaxTable$op == ":=",]) == 0) {
+      regressions <- syntaxTable[syntaxTable$op == "~",]
+      if (nrow(regressions) > 0) {
+        syntax <- .semEffectsSyntax(originalSyntax, syntaxTable, regressions, dataset, options)
+      }
+    }
+
+    if (exists("syntax")) {
+      lav_args[["model"]] <- syntax
+    } else {
+      lav_args[["model"]] <- originalSyntax
+    }
```

Comment:
So I am not certain why we have these changes, but they make the tests fail... Maybe this is not as easy as I thought...